### ISSUE TITLE
why is the trained images in yellow?

### ISSUE THREAD

**Issue by frozen-mind** [vinesmsuic/White-box-Cartoonization-PyTorch#8]
![image](https://user-images.githubusercontent.com/56190871/196326392-650e5eaa-067a-44bc-a23e-f5975074ea88.png)
as you can see, the sky of this image is somehow in yellow, I searched on google and find nothing useful, sry to bother you again

**Comment by markrmiller** [comment]
I found that most of the color or artifact type issues where it differs from the tensorflow version come from the padding that is used. I believe there is comment where it uses zeros padding mode about that being tensorflows default, but the tf version actually uses tf-slim and that is using same padding, and of particular importance, the padding amount calculated for that differs from the padding used here.

I adjusted that to match and that may be all that's needed, it's certainly minimally needed, but I also added histogan loss, and the colors are perfect.

**Comment by frozen-mind** [comment]
Perfect, but did you pushed those changes in this repository? Cuz I didn't find them....

**Comment by markrmiller** [comment]
FWIW here is the method I use to calculate padding:
def calculate_same_padding(input_size, kernel_size, stride):
    if stride == 1:
        return (kernel_size - 1) // 2
    else:
        output_size = math.ceil(float(input_size) / float(stride))
        pad_total = max((output_size - 1) * stride + kernel_size - input_size, 0)
        return pad_total // 2

And histogan: https://github.com/mahmoudnafifi/HistoGAN

**Comment by vinesmsuic** [comment]
Maybe too late to reply but I think it could be due to data bias. The anime images come from Ghibli-style images. And Ghibli-style images often feel warm and nostalgic, and that "yellow tone" you’re noticing is a big part of it. 

Nowadays even people uses ChatGPT to draw their "Ghibli style" images it also tends to give a yellow color tone.
